From: =?utf-8?q?Mart=C3=ADn_Ferrari?= <ticho@debian.org>
Date: Wed, 6 Apr 2016 15:39:48 +0200
Subject: Fix middle click action in the photo frame

Avoid using unescaped file names by using spawn, store the path in order
to use it.
---
 .../usr/share/cinnamon/desklets/photoframe@cinnamon.org/desklet.js | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/files/usr/share/cinnamon/desklets/photoframe@cinnamon.org/desklet.js b/files/usr/share/cinnamon/desklets/photoframe@cinnamon.org/desklet.js
index 79c415d..21820e5 100644
--- a/files/usr/share/cinnamon/desklets/photoframe@cinnamon.org/desklet.js
+++ b/files/usr/share/cinnamon/desklets/photoframe@cinnamon.org/desklet.js
@@ -213,6 +213,7 @@ MyDesklet.prototype = {
 
             let old_pic = this.currentPicture;
             this.currentPicture = image;
+            this.currentPicture.path = image_path;
 
             if (this.fade_delay > 0) {
                 Tweener.addTween(this._bin,
@@ -245,9 +246,9 @@ MyDesklet.prototype = {
             if (event.get_button() == 1) {
                 this._update();
             }
-            else if (event.get_button() == 2) {     
-                Util.spawnCommandLine("xdg-open " + this.currentPicture);                
-            }            
+            else if (event.get_button() == 2) {
+                Util.spawn(["xdg-open", this.currentPicture.path]);
+            }
         }
         catch (e) {
             global.logError(e);
